lib/managers/files.js

```diff
@@ -1243,6 +1251,78 @@ Files.prototype.getCollaborations = function(fileID, options, callback) {
 };
 
 /**
+ * Enum of valid x-rep- hint values for generating representation
+ *
+ * @readonly
+ * @enum {FileRepresentationType}
+ */
+Files.prototype.representation = {
+	PDF: '[pdf]',
+	THUMBNAIL: '[jpg?dimensions=320x320]',
+	IMAGE_MEDIUM: '[jpg?dimensions=1024x1024][png?dimensions=1024x1024]',
+	IMAGE_LARGE: '[jpg?dimensions=2048x2048][png?dimensions=2048x2048]',
+	EXTRACTED_TEXT: '[extracted_text]'
+};
+/**
+ * Requests links to all representations generated for a specific Box file
+ *
+ * API Endpoint: '/files/:fileID?fields=representations'
+ * Method : GET
+ *
+ * @param {string} fileID - Box ID of the file being requested
+ * @param {string} representationType - The x-rep-hints value the application should create a
+ *    representation for. This value can either come from FileRepresentationType enum or manually created
+ * @param {Object} [options] - Additional options. Can be left null in most cases.
+ * @param {string} [options.set_content_disposition_type] - Set to either 'inline' for the file to be opened in a browser tab or 'attachment'
+ * for the representation to be downloaded from the browser. If not specified the default behavior is to allow the user-agent of the client
+ * to determine how it should handle the returned representation object. To ensure consistency, we recommend supplying this parameter
+ * @param {string} [options.set_content_disposition_filename] - If defined this will name the downloaded representation file, if not the name will
+ *    be the original Box file name
+ * @param {string} [options.asset_path] - This optional field is for requesting a multi-page representation. This value should be in format of
+			<page_number>.<png>. 4-page long document will be named: 1.png, 2.png, etc...
+ * @param {Function} [callback] - A representation object containing the url_template and asset_url of the representation
+ * @returns {Promise<Object>} A promise resolving to the representation response objects
+*/
+Files.prototype.getRepresentationInfo = function(fileID, representationType, options, callback) {
+	var assetPath = options.asset_path || '';
+	delete options.asset_path;
+
+	var params = {
+		qs: {
+			fields: 'representations'
+		},
+		headers: {
+			'x-rep-hints': representationType
+		}
+	};
+	Object.assign(params.headers, options);
+	var apiPath = urlPath(BASE_PATH, fileID);
+
+	return this.client.get(apiPath, params)
+		.then(response => {
+			switch (response.statusCode) {
+				// 202 - A Box file representation will be generated, but is not ready yet
+			case httpStatusCodes.ACCEPTED:
```

Comment:
[style] These should still be indented to line up with the comments on the line above